test/envoye2e/http_metadata_exchange/http_metadata_exchange_test.go

```diff
@@ -0,0 +1,274 @@
+
+const tlsContext = `tls_context:
+  common_tls_context:
+    alpn_protocols:
+    - istio2
+    tls_certificates:
+    - certificate_chain: { filename: "testdata/certs/cert-chain.pem" }
+      private_key: { filename: "testdata/certs/key.pem" }
+    validation_context:
+      trusted_ca: { filename: "testdata/certs/root-cert.pem" }
+  require_client_certificate: true`
+
+const clusterTLSContext = `tls_context:
+  common_tls_context:
+    alpn_protocols:
+    - istio2
+    tls_certificates:
+    - certificate_chain: { filename: "testdata/certs/cert-chain.pem" }
+      private_key: { filename: "testdata/certs/key.pem" }
+    validation_context:
+      trusted_ca: { filename: "testdata/certs/root-cert.pem" }`
+
+const outboundNodeMetadata = `"NAMESPACE": "default",
+"INCLUDE_INBOUND_PORTS": "9080",
+"app": "productpage",
+"EXCHANGE_KEYS": "NAME,NAMESPACE,INSTANCE_IPS,LABELS,OWNER,PLATFORM_METADATA,WORKLOAD_NAME,CANONICAL_TELEMETRY_SERVICE,MESH_ID,SERVICE_ACCOUNT",
+"INSTANCE_IPS": "10.52.0.34,fe80::a075:11ff:fe5e:f1cd",
+"pod-template-hash": "84975bc778",
+"INTERCEPTION_MODE": "REDIRECT",
+"SERVICE_ACCOUNT": "bookinfo-productpage",
+"CONFIG_NAMESPACE": "default",
+"version": "v1",
+"OWNER": "kubernetes://apis/apps/v1/namespaces/default/deployments/productpage-v1",
+"WORKLOAD_NAME": "productpage-v1",
+"ISTIO_VERSION": "1.3-dev",
+"kubernetes.io/limit-ranger": "LimitRanger plugin set: cpu request for container productpage",
+"POD_NAME": "productpage-v1-84975bc778-pxz2w",
+"istio": "sidecar",
+"PLATFORM_METADATA": {
+ "gcp_cluster_name": "test-cluster",
+ "gcp_project": "test-project",
+ "gcp_cluster_location": "us-east4-b"
+},
+"LABELS": {
+ "app": "productpage",
+ "version": "v1",
+ "pod-template-hash": "84975bc778"
+},
+"ISTIO_PROXY_SHA": "istio-proxy:47e4559b8e4f0d516c0d17b233d127a3deb3d7ce",
+"NAME": "productpage-v1-84975bc778-pxz2w",`
+
+const inboundNodeMetadata = `"NAMESPACE": "default",
+"INCLUDE_INBOUND_PORTS": "9080",
+"app": "ratings",
+"EXCHANGE_KEYS": "NAME,NAMESPACE,INSTANCE_IPS,LABELS,OWNER,PLATFORM_METADATA,WORKLOAD_NAME,CANONICAL_TELEMETRY_SERVICE,MESH_ID,SERVICE_ACCOUNT",
+"INSTANCE_IPS": "10.52.0.34,fe80::a075:11ff:fe5e:f1cd",
+"pod-template-hash": "84975bc778",
+"INTERCEPTION_MODE": "REDIRECT",
+"SERVICE_ACCOUNT": "bookinfo-ratings",
+"CONFIG_NAMESPACE": "default",
+"version": "v1",
+"OWNER": "kubernetes://apis/apps/v1/namespaces/default/deployments/ratings-v1",
+"WORKLOAD_NAME": "ratings-v1",
+"ISTIO_VERSION": "1.3-dev",
+"kubernetes.io/limit-ranger": "LimitRanger plugin set: cpu request for container ratings",
+"POD_NAME": "ratings-v1-84975bc778-pxz2w",
+"istio": "sidecar",
+"PLATFORM_METADATA": {
+ "gcp_cluster_name": "test-cluster",
+ "gcp_project": "test-project",
+ "gcp_cluster_location": "us-east4-b"
+},
+"LABELS": {
+ "app": "ratings",
+ "version": "v1",
+ "pod-template-hash": "84975bc778"
+},
+"ISTIO_PROXY_SHA": "istio-proxy:47e4559b8e4f0d516c0d17b233d127a3deb3d7ce",
+"NAME": "ratings-v1-84975bc778-pxz2w",`
```

Comment:
nit: please don't embed config in sources. They are difficult to upgrade when the config needs to be changed.

Reply:
Agreed.. there is a lot of common config in tests.. will update them all together into templates...

Reply:
I started dumping them into https://github.com/istio/proxy/tree/master/testdata. You just need to assemble them as necessary.

Reply:
will do that, in a cleanup cl following this PR